tests/query/processors/test_arrayjoin_optimizer.py

```diff
@@ -429,19 +431,24 @@ def parse_and_process(snql_query: str) -> ClickhouseQuery:
             get_app_id("blah"), {"tenant_type": "tenant_id"}, "blah", None, None, None
         ),
     )
-    entity = get_entity(query.get_from_clause().key)
+    from_clause = query.get_from_clause()
+    assert isinstance(from_clause, QueryEntity)
+    entity = get_entity(from_clause.key)
     storage = entity.get_writable_storage()
     assert storage is not None
+    assert isinstance(query, LogicalQuery)
     for p in entity.get_query_processors():
         p.process_query(query, request.query_settings)
 
-    ArrayJoinKeyValueOptimizer("tags").process_query(query, request.query_settings)
```

Comment:
When I made the change, this test started failing.

The `ArrayJoinKeyValueOptimizer` is a storage level processor, but in this test it was being called before the query even knew what storage it was using. This only worked because the `TagsExpanderProcessor` was an entity processor.  Moving this line below the plan builder (where it would actually be called) fixed the tests.